decisions/02-timeline-functionality.md

```diff
@@ -0,0 +1,69 @@
+# Timeline control for animated design functions
+
+## Status
+
+Proposed
+
+## Context
+
+When fine tweaking values for an animated design function developers might find
+themselves in a situation where they just a need a single frame of the animation
+for reference. They want to be able to update an input value or a constant in
+code and judge the visual effect of this change based on a single frame.
+
+We had this experience when using Mechanic to develop the title and credit
+sequences for a movie.
+
+Using the current setup there are ways of manually adding this behavior, but
+they require developers to work with their own drawloops, add the needed input
+and take care of the conditional rendering (animate vs. still frame) themselves.
+
+This document proposes a timeline input as part of mechanic core. It is intended
+as an opt-in developer feature that's most useful in purely frame based
+animations. It makes heavy use of the [proposed](https://github.com/designsystemsinternational/mechanic/pull/156) new [Animation API](https://github.com/designsystemsinternational/mechanic/pull/157).
+
+## Decision
+
+We propose a timeline input that developers can opt into for animated design
+functions. To do so they'd have to set the `showTimeline` property of their
+function's settings to `true`.
+
+By default the timeline renders with two additional number inputs for the min
+frame number and the max frame number. The timeline scrubber interpolates
+between these two values. This makes sure a developer can zoom in and zoom out
+of the animation area they need.
+
+Alternatively developers can hardcode the maximum number of frames for the
+timeline by passing the `timelineMaxFrames` setting. If this is set, the timline
+input will just show the scrubber and interpolate between `0 ` and the value of
+`timelineMaxFrames`.
+
+The timeline is a development feature only. When building your design tool for
+deployment it is not exported.
```

Comment:
I'm not entirely sure about this. I can easily see users wanting to preview certain frames of their exports based on their input settings.

Reply:
Agreed!

Reply:
Ok, what would that mean for the export then? Would you be able to export only the range of the timeline then? Does setting the timeline to one frame allow you to export a single frame then?

Reply:
I would argue that the setting of the timeline has no affect on the export. The export exports the whole thing no matter the setting on the timeline. This is also how it works in After Effects. You can zoom, but an export is the whole thing.

Reply:
Same here! I understood Rune's initial comment about the deployment of the tool. In a dev or prod scenario the timeline could be useful.